src/main/java/mServer/crawler/AddToFilmlist.java

```diff
@@ -5,176 +5,270 @@
-                    Log.sysLog("===== Liste einsortieren: ABBRUCH =====");
-                    Log.sysLog("COUNT_MAX erreicht [s]: " + MAX_WAIT_TIME * 2);
-                    Log.sysLog("");
-                    stopOld = true;
+                count++;
+                if (count % 5 == 0) {
+                    long curSize = listeEinsortieren.size();
+                    System.out.println("Liste: " + curSize);
+                    System.out.println("Entfernte Einträge: " + ((oldSize - curSize)));
+                    oldSize = curSize;
                 }
-                wait(2000);
+                TimeUnit.SECONDS.sleep(2);
             } catch (Exception ex) {
                 Log.errorLog(978451205, ex, "Fehler beim Import Old");
             }
         }
 
-        vonListe.addAll(filteredOnline);
+        stopThreads();
+
+        final int treffer = retrieveThreadResults();
 
         Log.sysLog("===== Liste einsortieren: Noch online =====");
         Log.sysLog("Liste einsortieren, Anzahl: " + size);
-        Log.sysLog("Liste einsortieren, entfernt: " + (size - treffer.get()));
+        Log.sysLog("Liste einsortieren, entfernt: " + (size - treffer));
         Log.sysLog("");
-        Log.sysLog("In Liste einsortiert: " + treffer.get());
+        Log.sysLog("In Liste einsortiert: " + treffer);
         Log.sysLog("");
-        return treffer.get();
+        return treffer;
     }
 
-    private boolean stopOld = false;
+    /**
+     * 
+     * Add all local thread results to the filmlist.
+     *
+     * @return the total number of entries found.
+     */
+    private int retrieveThreadResults() {
+        int treffer = 0;
+        for (ImportOldFilmlistThread t : threadList) {
+            final ArrayList<DatenFilm> localList = t.getLocalAddList();
+            if (MserverDaten.debug)
+                Log.sysLog("Thread " + t.getName() + " list size: " + localList.size());
+            vonListe.addAll(localList);
+            localList.clear();
+            treffer += t.getTreffer();
+        }
+        return treffer;
+    }
 
-    private class AddOld implements Runnable {
+    private class ImportOldFilmlistThread extends Thread {
 
-        private DatenFilm film;
         private final ListeFilme listeOld;
-        private final int MIN_SIZE_ADD_OLD = 5; //REST eh nur Trailer
+        private final ArrayList<DatenFilm> localAddList = new ArrayList<>((vonListe.size() / NUMBER_OF_THREADS) + 500);
+        private int treffer = 0;
+        private OkHttpClient client = null;
 
-        public AddOld(ListeFilme listeOld) {
+        public ImportOldFilmlistThread(ListeFilme listeOld, OkHttpClient client) {
             this.listeOld = listeOld;
             threadCounter.incrementAndGet();
+            this.client = client;
+        }
+
+        public int getTreffer() {
+            return treffer;
+        }
+
+        public ArrayList<DatenFilm> getLocalAddList() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1319)